Fixed #29193 -- Fixed extra FK drops when altering unique fields.

jmbowman opened the pull request:
https://code.djangoproject.com/ticket/29193

carltongibson commented on tests/migrations/test_operations.py:
```diff
@@ -1402,6 +1403,33 @@ def test_rename_field_reloads_state_on_fk_target_changes(self):
+        AlterField will inadvertently drop and recreate the Pony.rider
+        foreign key even though it doesn't restrict the field being changed.
+        """
+        # Regression test for #28305
```
This could be part of the docstring.

carltongibson commented on django/db/backends/base/schema.py:
```diff
@@ -14,12 +14,28 @@
+    on its model from the given relation?
+    """
+    field = relation.field
+    if field.many_to_many:
```
Confirm that. (An `assert False` thrown in here is never triggered.)

jmbowman commented:
@timgraham @carltongibson I've moved the regression test (and updated it accordingly), moved the ticket number comment to the docstring, and added a second commit with an extra test case to cover the reverse M2M relation case.  Let me know if you see anything else that could use improvement.
